### PR TITLE
V3.0.15 carto updated cached marker rendering [WIP]

### PR DESCRIPTION
**Pull request by rafatower**

### CODE REVIEW

**Comment by Algunenano** [comment]
I've clean up the empty commits and picked @mtehver fix that broke OSX and everything seems fine. I've relaunched the OSX build a couple of times (manually from Travis) and it seems ok now.

File: .travis.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ matrix:
     - os: osx
       compiler: ": clang-osx"
       # https://docs.travis-ci.com/user/languages/objective-c/#Supported-OS-X-iOS-SDK-versions
-      osx_image: xcode7.3 # upgrades clang from 6 -> 7
+      osx_image: xcode9.2 # this seems to be in sync with Linux clang
```
Comment (rafatower) [inline .travis.yml]: this change is not actually needed. I'd avoid it

File: .travis.yml [hunk 2]
```diff
@@ -108,4 +108,4 @@ script:
  # we allow visual failures with g++ for now: https://github.com/mapnik/mapnik/issues/3567
  - if [[ ${RESULT} != 0 ]] && [[ ${CXX} =~ 'clang++' ]]; then false; fi;
  - enabled ${COVERAGE} coverage
- - enabled ${BENCH} make bench
\ No newline at end of file
+ - enabled ${BENCH} make bench - enabled ${BENCH} make bench - enabled ${BENCH} make bench
```
Comment (rafatower) [inline .travis.yml]: same here

**Review by rafatower — COMMENTED** [review]
@Algunenano some commits are not needed at all. They may call for conflicts eventually and that is why I cleaned up many of them in this other branch (`git rebase -i ...`):

https://github.com/CartoDB/mapnik/tree/v3.0.15-carto.rebased-cached-marker-rendering

https://github.com/CartoDB/mapnik/compare/v3.0.15-carto...CartoDB:v3.0.15-carto.rebased-cached-marker-rendering?expand=1

If you take a look, the commit history is cleaner there.

(sorry, I forgot to push `rebased-cached-marker-rendering` when I found the osx caching issue)

In the end it is up to you what to use or whether rewrite history or not.

**Comment by Algunenano** [comment]
I've clean up a bit the PR to leave just 3 commits:
- New functionality.
- Update test branch.
- Benchmark file.

I'll run some tests and if everything goes OK I'll merge it and start testing performance locally.

**Pull request merged**